storage/src/vespa/storage/visiting/visitormanager.cpp

```diff
@@ -208,39 +202,33 @@ VisitorManager::run(framework::ThreadHandle& thread)
     while (true) {
         thread.registerTick(framework::PROCESS_CYCLE);
         {
-            vespalib::LockGuard waiter(_visitorLock);
+            std::lock_guard waiter(_visitorLock);
             if (thread.interrupted()) {
                 break;
             }
             timedOut = _visitorQueue.releaseTimedOut();
         }
-        framework::MicroSecTime currentTime(
-                _component.getClock().getTimeInMicros());
+        framework::MicroSecTime currentTime(_component.getClock().getTimeInMicros());
         for (std::list<CQ::CommandEntry>::iterator it = timedOut.begin();
              it != timedOut.end(); ++it)
         {
-            _metrics->queueTimeoutWaitTime.addValue(
-                    currentTime.getTime() - it->_time);
-            std::shared_ptr<api::StorageReply> reply(
-                    it->_command->makeReply().release());
-            reply->setResult(api::ReturnCode(api::ReturnCode::BUSY,
-                                        "Visitor timed out in visitor queue"));
+            _metrics->queueTimeoutWaitTime.addValue(currentTime.getTime() - it->_time);
+            std::shared_ptr<api::StorageReply> reply(it->_command->makeReply());
+            reply->setResult(api::ReturnCode(api::ReturnCode::BUSY,"Visitor timed out in visitor queue"));
             sendUp(reply);
         }
         {
-            vespalib::MonitorGuard waiter(_visitorLock);
+            std::unique_lock waiter(_visitorLock);
             if (thread.interrupted()) {
                 break;
             } else if (_visitorQueue.empty()) {
-                waiter.wait(1000);
+                _visitorCond.wait_for(waiter, 1000ms);
                 thread.registerTick(framework::WAIT_CYCLE);
             } else {
-                uint64_t timediff = (_visitorQueue.tbegin()->_time
-                                     - currentTime.getTime())
-                                            / 1000000;
+                uint64_t timediff = (_visitorQueue.tbegin()->_time- currentTime.getTime()) / 1000000;
                 timediff = std::min(timediff, uint64_t(1000));
                 if (timediff > 0) {
-                    waiter.wait(timediff);
+                    _visitorCond.wait_for(waiter, std::chrono::milliseconds(timediff));
```

Comment:
I'm a bit puzzled by the original code here. Since the queued timestamp and `currentTime` are both in microseconds, dividing by 1,000,000 implies that `timediff` is in discrete seconds and not in milliseconds. Expected wait period will therefore be off by a few orders of magnitude... I don't think this has mattered in practice, since it just means we'll wake up the thread more frequently than strictly required.

Reply:
Then we are 2. I think it is overdue to replace replace the framework times with vespalib::duration. I think that was one of things we postponed last time we tried to unify time.